code/go/0chain.net/blobbercore/handler/storage_handler.go

```diff
@@ -667,6 +669,118 @@ func (fsh *StorageHandler) GetObjectTree(ctx context.Context, r *http.Request) (
 	return &refPathResult, nil
 }
 
+//Retrieves file refs. One can use three types to refer to regular, updated and deleted. Regular type gives all undeleted rows.
+//Updated gives rows that is updated compared to the date given. And deleted gives deleted refs compared to the date given.
+func (fsh *StorageHandler) GetRefs(ctx context.Context, r *http.Request) (*blobberhttp.RefResult, error) {
+	allocationTx := ctx.Value(constants.ALLOCATION_CONTEXT_KEY).(string)
+	allocationObj, err := fsh.verifyAllocation(ctx, allocationTx, false)
+
+	if err != nil {
+		return nil, common.NewError("invalid_parameters", "Invalid allocation id passed."+err.Error())
+	}
+
+	clientSign, _ := ctx.Value(constants.CLIENT_SIGNATURE_HEADER_KEY).(string)
+	valid, err := verifySignatureFromRequest(allocationTx, clientSign, allocationObj.OwnerPublicKey)
+	if !valid || err != nil {
+		return nil, common.NewError("invalid_signature", "Invalid signature")
+	}
+
+	allocationID := allocationObj.ID
+	clientID := ctx.Value(constants.CLIENT_CONTEXT_KEY).(string)
+	if len(clientID) == 0 || allocationObj.OwnerID != clientID {
+		return nil, common.NewError("invalid_operation", "Operation needs to be performed by the owner of the allocation")
+	}
+	path := r.FormValue("path")
+	if len(path) == 0 {
+		return nil, common.NewError("invalid_parameters", "Invalid path")
+	}
+
+	pageLimitStr := r.FormValue("pageLimit")
+	var pageLimit int
+	if len(pageLimitStr) == 0 {
+		pageLimit = PageLimit
+	} else {
+		o, err := strconv.Atoi(pageLimitStr)
+		if err != nil {
+			return nil, common.NewError("invalid_parameters", "Invalid page limit value type")
+		}
+		if o <= 0 {
+			return nil, common.NewError("invalid_parameters", "Zero/Negative page limit value is not allowed")
+		} else if o > PageLimit {
+			pageLimit = PageLimit
+		} else {
+			pageLimit = o
+		}
+	}
+	offsetPath := r.FormValue("offsetPath")
+	offsetDate := r.FormValue("offsetDate")
+	updatedDate := r.FormValue("updatedDate")
+	err = checkValidDate(offsetDate)
+	if err != nil {
+		return nil, err
+	}
+	err = checkValidDate(updatedDate)
+	if err != nil {
+		return nil, err
+	}
+	fileType := r.FormValue("fileType")
+	levelStr := r.FormValue("level")
+	var level int
+	if len(levelStr) != 0 {
+		level, err = strconv.Atoi(levelStr)
+		if err != nil {
+			return nil, common.NewError("invalid_parameters", err.Error())
+		}
+		if level < 0 {
+			return nil, common.NewError("invalid_parameters", "Negative level value is not allowed")
+		}
+	}
+
+	refType := r.FormValue("refType")
+	var refs *[]reference.Ref
+	var totalPages int
+	var newOffsetPath string
+	var newOffsetDate string
+
+	switch {
+	case refType == "regular":
```

Comment:
do you think it is better to implement them with restful style? regular = GET/updated=PUT/deleted=DELETE.

Reply:
I see them on `reference`. They mean `order by`. hmm, the naming is a bit confusing.

Reply:
do you think it is better to rename them 
- GetRefsOrderByUpdated
- GetRefsOrderByDeleted

Reply:
All the endpoints are actually GET request. We Get;
Regular refs: Those refs that are not deleted
Updated refs: Those refs that are updated compared to the given time reference
Deleted refs: Those refs that are deleted compared to the given time reference

So it doesn't make sense to use PUT and DELETE since they mean to update and delete the rows but here we only get the rows.

Reply:
got it.